File: UIMA.Ext.FSMMatching/src/main/scala/ru/kfu/itis/issst/uima/fsmm/nfsm/AnnotationMatcher.scala
/**
 *
 */
package ru.kfu.itis.issst.uima.fsmm.nfsm
import ru.kfu.itis.issst.uima.fsmm.pattern
import pattern.AttributeExtractor
import pattern.ValueMatcher

/**
 * @author Rinat Gareev (Kazan Federal University)
 *
 */
class AnnotationMatcher[A](
  val attributeExtractor: AttributeExtractor[A],
  val matcher: ValueMatcher, val expectedValue: Any) {

  def matches(anno: A, varCtx: VariableContext): (Boolean, VariableContext) = {
    val attrValue = attributeExtractor.getValue(anno)
    matcher.matches(attrValue.asInstanceOf[matcher.ActualType],
      expectedValue.asInstanceOf[matcher.TargetType], varCtx)
  }

}

object AnnotationMatcher {
}

File: UIMA.Ext.PosTagger.TCRF/src/test/java/ru/kfu/itis/issst/uima/test/AnnotationMatcher.java
package ru.kfu.itis.issst.uima.test;

import org.apache.uima.cas.text.AnnotationFS;
import org.mockito.ArgumentMatcher;

/**
 * @author Rinat Gareev
 */
class AnnotationMatcher<T> extends ArgumentMatcher<T> {

    private String expectedTxt;

    public AnnotationMatcher(String expectedTxt) {
        this.expectedTxt = expectedTxt;
    }

    @Override
    public boolean matches(Object arg) {
        if(arg == null) return false;
        AnnotationFS argAnno = (AnnotationFS) arg;
        return expectedTxt.equals(argAnno.getCoveredText());
    }
}
